packages/@aws-cdk/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -93,22 +94,28 @@ export class AthenaStartQueryExecution extends sfn.TaskStateBase {
 
     policyStatements.push(
       new iam.PolicyStatement({
-        actions: ['s3:AbortMultipartUpload',
-          's3:CreateBucket',
-          's3:GetBucketLocation',
-          's3:GetObject',
+        actions: ['s3:CreateBucket',
           's3:ListBucket',
+          's3:GetBucketLocation',
+          's3:GetObject'],
+        resources: ['*'], // Need * permissions to create new output location https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+      }),
+    );
+
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['s3:AbortMultipartUpload',
           's3:ListBucketMultipartUploads',
           's3:ListMultipartUploadParts',
           's3:PutObject'],
-        resources: [this.props.resultConfiguration?.outputLocation ?? '*'], // Need S3 location where data is stored https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+        resources: [this.props.resultConfiguration?.outputLocation?.bucketName ? `arn:aws:s3:::${this.props.resultConfiguration?.outputLocation?.bucketName}/${this.props.resultConfiguration?.outputLocation?.objectKey}/*` : '*'], // Need S3 location where data is stored or Athena throws an Unable to verify/create output bucket https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
```

Comment:
@Sumeet-Badyal Thank you for your PR!
I've tried the changes in this PR, but since `objectKey` is required, it doesn't seem to work when specifying into directly under the bucket.
Another problem might be that when using [CREATE TABLE AS](https://docs.aws.amazon.com/athena/latest/ug/create-table-as.html), files are created under `tables/` directory of the bucket specified with `outputLocation`, so we have to specify `arn:aws:s3:::my-bucket/*` (not including any object keys except wildcard) instead of `arn:aws:s3:::my-bucket/foo/*`.